compiler/modules/front.syntax/tests/syntaxNode.bal

```diff
@@ -0,0 +1,566 @@
+function syntaxNodeFromImportDecl(ImportDecl decl) returns NonTerminalSyntaxNode {
+    SyntaxNode[] childNodes = [];
+    childNodes.push();
+    string? org = decl.org;
+    string? prefix = decl.prefix;
+    return finishWithSemiColon(decl,
+                               { token: "import", pos: decl.startPos },
+                               org != () ? childNodes.push({ name: org, pos: () }) : (),
+                               org != () ? childNodes.push({ token: "/" }) : (),
+                               { name: ".".'join(...decl.names), pos: decl.namePos },
+                               prefix != () ? childNodes.push({ token: "as" }) : (),
+                               prefix != () ? childNodes.push({ name: prefix, pos: () }) : ());
+}
+
+function syntaxNodeFromModuleLevelDefn(ModuleLevelDefn defn) returns SyntaxNode {
+    if defn is ConstDefn {
+        return syntaxNodeFromConstDefn(defn);
+    }
+    else if defn is TypeDefn {
+        return syntaxNodeFromTypeDefn(defn);
+    }
+    else {
+        return syntaxNodeFromFunctionDefn(defn);
+    }
+}
+
+function syntaxNodeFromConstDefn(ConstDefn defn) returns SyntaxNode {
+    SubsetBuiltinTypeDesc? td = defn.td;
+    return finishWithSemiColon(defn,
+                               { token: "const" },
+                               td != () ? { token: td.builtinTypeName } : (),
+                               { name: defn.name, pos: defn.namePos },
+                               { token: "=" },
+                               syntaxNodeFromExpr(defn.expr)
+    );
+}
+
+function syntaxNodeFromTypeDefn(TypeDefn defn) returns SyntaxNode {
+    return finishWithSemiColon(defn,
+                               { token: "type" },
+                               { name: defn.name, pos: defn.namePos },
+                               syntaxNodeFromTypeDesc(defn.td));
+}
+
+function syntaxNodeFromFunctionDefn(FunctionDefn defn) returns SyntaxNode {
+    return nonTerminalSyntaxNode(defn,
+                                 { token: "function" },
+                                 { name: defn.name, pos: defn.namePos},
+                                 syntaxNodeFromFunctionTypeDesc(defn.typeDesc, skipHeader = true),
+                                 syntaxNodeFromStmtBlock(defn.body));
+}
+
+function syntaxNodeFromFunctionTypeParam(FunctionTypeParam param) returns SyntaxNode {
+    string? name = param.name;
+    return nonTerminalSyntaxNode(param,
+                                 syntaxNodeFromTypeDesc(param.td),
+                                 name != () ? { name, pos: param.namePos} : ());
+}
+
+function syntaxNodeFromStmtBlock(StmtBlock block) returns NonTerminalSyntaxNode {
+    return nonTerminalSyntaxNode(block, ...combineChildNodes([{ token: "}", pos: block.startPos}],
```

Comment:
Don't see why you need combineChildNodes.

Maybe allow params of nonTerminalSyntaxNode to be lists.

Reply:
fixed